org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/actions/CompareObjectsAction.java

```diff
@@ -0,0 +1,445 @@
+package org.eclipse.jdt.internal.debug.ui.actions;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.jdt.debug.core.IJavaVariable;
+import org.eclipse.jdt.internal.debug.ui.ObjectComparision;
+import org.eclipse.jface.action.IAction;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.jface.viewers.ISelectionProvider;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.ui.IWorkbenchPart;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.IWorkbenchWindowActionDelegate;
+
+public class CompareObjectsAction extends ObjectActionDelegate implements IWorkbenchWindowActionDelegate {
+	protected IWorkbenchWindow fWindow;
+	private static final String TITLE_NAME_SUCCESS = "Object Comparision"; //$NON-NLS-1$
+	private static final String TITLE_NAME_FAILED = "Object Comparision Failed"; //$NON-NLS-1$
+
+	private static final int DISPLAY_LIMIT = 20;
+	@Override
+	public void init(IWorkbenchWindow window) {
+		this.fWindow = window;
+	}
+
+	@Override
+	protected IWorkbenchPart getPart() {
+		IWorkbenchPart part = super.getPart();
+		if (part != null) {
+			return part;
+		} else if (fWindow != null) {
+			return fWindow.getActivePage().getActivePart();
+		}
+		return null;
+	}
+
+	@Override
+	public void run(IAction action) {
+		if (getPart() != null) {
+			ISelectionProvider provider = getPart().getSite().getSelectionProvider();
+			if (provider != null) {
+				ISelection selection = provider.getSelection();
+				if (selection instanceof IStructuredSelection selections) {
+					compareSelectedObjects(selections.toList());
+					return;
+				}
+			}
+		}
+	}
+
+	/**
+	 * Parent method for comparing selected two objects from variable view
+	 *
+	 * @param selections
+	 *            Selected two objects from variable view
+	 */
+	@SuppressWarnings("nls")
+	protected void compareSelectedObjects(List<IStructuredSelection> selections) {
+
+		Map<String, Object> resultMap = new HashMap<>();
+		if (selections != null && !selections.isEmpty()) {
+			IJavaVariable selectedObject1 = null;
+			IJavaVariable selectedObject2 = null;
+			if (selections.get(0) instanceof IJavaVariable && selections.get(1) instanceof IJavaVariable) {
+				selectedObject1 = ((IJavaVariable) selections.get(0));
+				selectedObject2 = ((IJavaVariable) selections.get(1));
+			}
+			if (selectedObject2 != null && selectedObject1 != null) {
+				ObjectComparision obcp = new ObjectComparision();
+				try {
+					String referenceTypeObject1 = selectedObject1.getValue().getReferenceTypeName();
+					String referenceTypeObject2 = selectedObject2.getValue().getReferenceTypeName();
+
+					if (referenceTypeObject1.equals(referenceTypeObject2)) {
+
+						String selectionReference = selectedObject1.getReferenceTypeName();
+						if (selectionReference.equals("java.lang.String")) {
```

Comment:
can you use interface CharSequence instead of String implementation?

Reply:
from getReferenceTypeName() and javaType() only "java.lang.String" is getting returned.